pdf-2/src/main/java/com/baeldung/exceltopdf/ExcelToPDFConverter.java

```diff
@@ -74,9 +88,27 @@ private static void addTableHeader(XSSFSheet worksheet, PdfPTable table) throws
         Row headerRow = worksheet.getRow(0);
         for (int i = 0; i < headerRow.getPhysicalNumberOfCells(); i++) {
             Cell cell = headerRow.getCell(i);
-            String headerText = cell.getCellType() == CellType.STRING ? cell.getStringCellValue() : String.valueOf(cell.getNumericCellValue());
+
+            String headerText;
+            switch (cell.getCellType()) {
+            case STRING:
+                headerText = cell.getStringCellValue();
+                break;
+            case NUMERIC:
+                headerText = String.valueOf(BigDecimal.valueOf(cell.getNumericCellValue()));
+                break;
+            case BLANK:
+                headerText = ""; // or null
+                break;
+            default:
+                logger.warn("Unsupported cell type: {}", cell.getCellType());
```

Comment:
Maybe it's worth mentioning in the article that an Excel file that contains formulae isn't fully supported.

Reply:
Okay noted.